### PR TITLE
Edit Renter Candidate Page

### LINKED ISSUE #103: Edit Renter Candidate Flow
# Overview

Implement the page and process when the "Edit" button is pressed on the renter candidate page.

## Frontend

Make all the info fields of the renter candidate into editable text boxes and format it accordingly. Change the housing locator and status columns in the referral table to dropdowns. Add the popups when deleting a referral.

## Backend

Create the put route for renter candidates and connect it to the frontend.

Create the delete route for referrals and connect it to the frontend.

## Acceptance Criteria
- [x] Frontend page looks according to Figma, with input boxes and dropdowns in the right place
- [x] Successfully and accurately updates all renter candidate fields when the "Save Changes" button is pressed
- [x] Popup and referral deletion is supported

### PR DESCRIPTION
**Pull request by Azhou2023**
## Tracking Info

Resolves #103 

## Changes

Added the edit renter candidate process flow after pressing the "Edit" button. Allows changes to be made to all renter candidate info fields as well as to their current referrals. Implemented deletion for referrals. 

Also implemented a max number of 10 photos and 2 videos per unit, as well as a 200MB size cap for uploaded videos.

## Testing

Tested every edit field thoroughly, checked that deletion worked in backend and reflected in frontend. Checked for photo/video uploading that uploading capped with correct criteria (in both normal uploading and uploading at unit creation).

## Confirmation of Change

Screenshot 2024-05-23 at 10 29 24 PM
Screenshot 2024-05-23 at 10 29 36 PM
Screenshot 2024-05-23 at 10 29 43 PM
Screenshot 2024-05-23 at 10 29 52 PM
Screenshot 2024-05-23 at 10 31 22 PM

### CODE REVIEW

**Review by PoliteUnicorn — APPROVED** [review]
This looks good to me!

File: frontend/src/components/ListingForm/ImagesVideos.tsx [hunk 1]
```diff
@@ -88,6 +100,7 @@ export const ImagesVideos = ({ unit_id, onChange }: ImagesVideosProps) => {
     getFileMetadata(unit_id, "videos")
       .then((data) => {
         setAllVideos(data);
+        console.log(data);
```
Comment (petabite) [inline frontend/src/components/ListingForm/ImagesVideos.tsx]:
```suggestion
```

File: frontend/src/pages/RenterCandidatePage.tsx [hunk 2]
```diff
@@ -158,66 +345,298 @@ export function RenterCandidatePage() {
                 alt={"Back arrow"}
                 style={{ marginRight: "12px" }}
               />
-              Back
+              Back to Listing
```
Comment (petabite) [inline frontend/src/pages/RenterCandidatePage.tsx]: Users will be able to access the Renter Candidate page from the All Renter Candidate page (once it's implemented) as well. Thoughts on calling this "Back" to keep it more general?
```suggestion
              Back 
```

**Review by petabite — CHANGES_REQUESTED** [review]
Nice work! 💯 Left some tiny comments below, but should be good to go once you add the fixes we discussed in the team meeting!

**Review by petabite — APPROVED** [review]
Good stuff 🔥

**Pull request merged**